lib/fuel-rest.js

```diff
@@ -130,6 +130,7 @@ FuelRest.prototype._makeRequest = function(consolidatedOpts, callback) {
 		// check if we should retry req
 		if(helpers.isValid401(res) && consolidatedOpts.retry) {
 			requestOptions.auth = consolidatedOpts.auth;
+			this.AuthClient.invalidateToken();
```

Comment:
If the other PR has this, it's a bug; this needs to be `this.AuthClient.invalidateToken(accessToken);` and above this, we need to do `var accessToken = authResponse.accessToken` before the callback and also only if `options.headers.Authorization` was not provided.

@nathan-boyd